### PR TITLE
Release 0.4.0-rc2

### PR DESCRIPTION
**Pull request by ax3l**
Collection of all relevant PRs for the `release-0.4.0` branch in the next days/weeks.
Update to `0.4.0-rc2`.

## Bug Fixes

- [x] fix laser init on cpu #2689
- [x] fix out of host memory during HDF5 dumping #2690

## Misc

- [x] bump version to `0.4.0-rc2`
- [x] update manual #2685

### CODE REVIEW

File: USAGE.rst [hunk 1]
```diff
@@ -46,19 +46,20 @@ Step-by-Step
...
+:ref:`Runtime (command line) arguments <usage-cfg>` are set in ``etc/picongpu/*.cfg`` files.
+These options do *not* require a re-compile when changed (e.g. simulation size, number of devices, plugins, ...).
 
 2. Compile Simulation
 """""""""""""""""""""
 
-In our input, ``.param`` files are build directly into the PIConGPU binary for performance reasons.
-Changing or initially adding those requires a compile.
+In our input, ``.param`` files are build directly into the PIConGPU binary for :ref:`performance reasons <usage-params-rationale>`.
+A compile is required after changing or initially adding those files.
```
Comment (PrometheusPi) [inline USAGE.rst]: This is a repetition of the content of line 51-53.

File: USAGE.rst [hunk 2]
```diff
@@ -80,31 +83,42 @@ While you are still in ``$HOME/picInputs/myLWFA``, start your simulation on one
...
+Note that we can use the same ``1.cfg`` file, your input set is *portable*.
+
+.. _usage-basics-create:
 
 pic-create
 """"""""""
 
 This tool is just a short-hand to create a new set of input files.
-It does a copy from an already existing set of input files (e.g. our examples or a previous simulation) and adds additional default files.
+It copies from an already existing set of input files (e.g. our examples or a previous simulation) and adds additional helper files.
```
Comment (PrometheusPi) [inline USAGE.rst]: The meaning of *helper* is not clear to me.

File: docs/source/usage/tbg.rst [hunk 3]
```diff
@@ -3,15 +3,30 @@
...
+.. sectionauthor:: Axel Huebl
 .. moduleauthor:: René Widera
 
-todo: explain idea and use case
+Our tool *template batch generator* (``tbg``) abstracts program runtime options from technical details of supercomputers.
+On a desktop PC, one can just execute a command interactively and instantaneously.
+Contrarily on a supercomputer, resources need to be shared between different users efficiently via *job scheduling*.
+Scheduling on today's supercomputers is usually done via *batch systems* that define various queues of resources.
+
+An unfortunate aspect about batch systems from a user's perspective is, that their usage varies a lot.
```
Comment (PrometheusPi) [inline docs/source/usage/tbg.rst]: grammar: please remove comma
optional: about -> of

File: docs/source/usage/tbg.rst [hunk 4]
```diff
@@ -3,15 +3,30 @@
...
+And naturally, different systems have different resources in queues that need to be described.
+
+PIConGPU runtime options are described in *configuration files* (``.cfg``).
+We abstract the description of queues, resource acquisition and job submission via *template files* (``.tpl``).
+For example, a ``.cfg`` file defines how many *devices* shall be used for computation, but a ``.tpl`` file calculates how many *physical nodes* will be requested.
```
Comment (PrometheusPi) [inline docs/source/usage/tbg.rst]: The `.tpl`file does not calculate  the number of nodes, it defines a distribution which is used by `tbg` to perform a calculation.

File: USAGE.rst (hunk 1 above)
Comment (ax3l) [reply USAGE.rst]: See long message below. You reviewed and merged this already in #2685.
If you have further suggestions now, please open a PR with your proposed changes to `dev`.

This backport, cherry-pick only PR is not the right place for this.

**Comment by ax3l** [comment]
@psychocoderHPC ready to apply these fixes to the `0.4.0` release branch.
I will write an announcement about the `0.4.0-rc2` updates when this is merged.

**Pull request merged**